# Задачи (1)



## Сплави

Металургична компания иска да произведе сплав от 40 процента олово (lead), 30 процента цинк (zinc) и 30 процента калай (tin). Компанията може да закупи различни количества от други сплави (A, B, ..., I), които да претопи и да произведе желания продукт. Цената на сплавите и тяхното съдържание на олово, цинк и калай (в проценти) са дадени в @tbl-alloys. Колко и от кои сплави трябва да закупи компанията, за да произведе желаната сплав на минимална цена?

1. Намерете оптималното решение в [Excel]({{ var githubRawPrefix}}excel/Alloys.xlsx)
2. Опростете задачата, като игнорирате изискванията за процент олово и цинк и използвате само сплавите B, C, H, и E. Намерете оптималното решение графично. [Desmos](https://www.desmos.com/calculator/9jqnpdf5lo)


In [1]:
#| label: tbl-alloys
#| tbl-cap: "Съдържание на олово, цинк и калай (в проценти) и цени на налични сплавите"

import pandas as pd

alloys = pd.read_excel("excel/Alloys.xlsx")
alloys

,Alloy,A,B,C,D,E,F,G,H,I,Desired
0,% Lead,10.0,10.0,40.0,60,30.0,30.0,30.0,50.0,20.0,30
1,% Zinc,10.0,30.0,50.0,30,30.0,40.0,20.0,40.0,30.0,30
2,% Tin,80.0,60.0,10.0,10,40.0,30.0,50.0,10.0,50.0,40
3,Cost USD/kg,4.1,4.3,5.8,6,7.6,7.5,7.3,6.9,7.3,Min


## Кредитен портфейл

Банка е в процеса на определяне на кредитната си политика за пет различни сектора с характеристики дадени в следната таблица:

| Вид кредит      | Лихва | Дял на несъбираеми кредити |
|-----------------|------:|---------------------------:|
| Потребителски   |  0.15 |                       0.10 |
| Автомобили      |  0.13 |                       0.07 |
| Недвижими имоти |  0.12 |                       0.03 |
| Земеделие       | 0.125 |                       0.05 |
| Търговия        |  0.10 |                       0.02 |

Банката има бюджет от общо 12 милиона лева, които може да използва за кредитиране си при определяне на политиката трябва да се съобрази със следните регулации:

1. Поне 40% от заемите трябва да са за земеделие или търговия.
2. Заеми за недвижими имоти трябва да са поне половината от общата сума на потребителски кредити, кредити свързани с автомобили и недвижими имоти.
3. Според регулаторни изисквания дялът на несъбираемите кредити не трябва да надвишава 4% от всички кредити. Несъбираемите кредити не носят лихви.

Какъв е оптималният (най-висока печалба) план за банката?

## Планиране на градско строителство

Да приемем, че София е изправена пред дефицит в местния бюджет и общинският съвет обмисля как да увеличи приходите от данък сгради чрез преустройства на общински имоти. Проектът се състои от две части -- разчистване на занемарени и повредени постройки и построяване на нови жилища.

1. В момента на планиране общината притежава 300 занемарени постройки, които могат да бъдат съборени. Всяка от тях освобождава 1000 кв. метра и събарянето й струва 3 000 лв. на постройка. 15 процента от освободеното пространство е предвидено за улици, тротоари и свободни  пространства.
2. На освободените парцели общината може да построи четири вида нови жилищни сгради: еднофамилни (300 кв. метра), двуфамилни (500 кв. метра), три-фамилни (700 кв. метра) и четири-фамилни къщи (900 кв. метра). Очакваните данъчни постъпления са съответно 1 000 лв., 1 700 лв., 2 400 лв. и 2 800 лв. на година.
3. Поне 20 процента от новите постройки трябва да са еднофамилни къщи, двуфамилните домове трябва да съставляват поне 20 процента, а три и четири-фамилните домове трябва (заедно) да са поне една четвърт от всички нови постройки.
4. Строителните разходи за новите домове са съответно 50 000 лв., 70 000 лв., 130 000 лв. и 160 000 лв.
5. Общината предвижда на финансира проекта чрез банков заем, който не може да надвишава 15 милиона лева.

Колко от всеки вид къщи трябва да планира да построи общината, така че да постигне възможно най-високи данъчни постъпления?

## Оптимизация на производство (рафинерия)

Рафинерия произвежда три вида безоловен бензин, които се различават по октановото си число (ON): A: обикновен ($ON \geq 87$), B: премиум ($ON \geq 89$) и C: супер ($ON \geq 92$). Преработката на суровия петрол обхваща три компонента.

1. В първият компонент суровият петрол се дестилира до петролна суровина (feedstock) с октаново число ON = 82. За производството на един барел суровина са нужни 5 барела суров петрол. Капацитетът на дестилационните съоръжения е 1 500 хиляди барела на ден.
2. Част от петролната суровина се обработва допълнително в термичен крекер, който произвежда бензин с ON = 98. От един барел суровина крекерът произвежда половин барел бензин. Капацитетът на крекера е ограничен до 200 000 барела петролна суровина на ден.
3. Последният елемент в рафинерията е смесител, който комбинира бензин от крекера и петролна суровина от дестилацията. Октановото число на смеси е приблизително равно на претегленото средно на октановите числа на компонентите им. Пример: от 1 барел с ON=90 и 2 барела с ON=100 ще се получат три барела смес с октаново число $(1\times 90 + 2 \times 100) / 3 \approx 96.6$
4. Търсенето на трите вида бензин е ограничено до съответно 50, 30, и 40 хиляди барела на ден.
5. Рафинерията оценява печалбата от барел бензин на съответно 6, 7 и 8 долара за трите вида бензин, които продава.

Съставете математически модел, на базата на който да препоръчате оптимален (най-висока печалба) производствен план на рафинерията.

Виж @fig-refinery за производствената схема на рафинерията.


```{mermaid}
%%| label: fig-refinery
%%| fig-cap: Производствена схема на рафинерията

flowchart LR
    A[Дестилация 5:1] -->|ON=82| B[Крекер 2:1]
    A --> |ON=82| C
    B -->|ON=98| C[Смесител 1:1]
    C --> D(ON=87)
    C --> E(ON=89)
    C --> F(ON=92)
```


## Управление на ресторант

Управител на ресторант установява, че през изминалия сезон е реализирана значително по-малка от очакваната печалба. Една от основните причини е неефективно управление на наетия персонал. 

При планиране на дейността за следващия период управителят се обръща към вас за съвет как да подобри дейността. Той ви предоставя следната информация: 

- ресторантът наема сервитьори с различно ниво на опит, като заплащането на сервитьори с опит над 1 година е 15 EUR/час, съответно за опис от 6 месеца до 1 година е 10 EUR/час, а за опит под 6 месеца е 8 EUR/час.
-  Салонът е организиран в три секции – в първата секция са разположени маси за малки компании (до 4 човека), във втората секция – маси за средни по големина компании (до 8 човека), а в третата секция масите са за големи компании (над 8 човека). 


След анализ на данни от предходни сезони вие установявате, че на час средно са били обслужвани по 40 малки, 30 средни и 25 големи маси. През следващия сезон не се очакват промени. Според оценката на двамата помощник-мениджъри, сервитьор с опит над 1 година успява да обслужи една малка маса за сумарно време от 5 минути, една средна маса – за 6 минути, а една голяма маса – за 7.5 минути. Аналогичните данни за сервитьор с опит между 6 месеца и 1 година са 7.5, 10 и 15 минути. Сервитьори с опит по-малък от 6 месеца обслужват само малки и средни маси, като съответното време е 15 минути за малка маса и 30 минути за средна маса.


Вашата задача е да минимизирате разходите за възнаграждения, изплатени на сервитьорите, при условие, че ресторантът винаги трябва да разполага с достатъчно на брой служители, които да обслужват заетите маси.